test/sshkit/ssh_test.exs

```diff
@@ -13,36 +13,150 @@ defmodule SSHKit.SSHTest do
 
   defmodule SSHSandboxError do
     def connect(_, _, _, _), do: {:error, :sandbox}
+
+    def close(_) do
+      send self(), :closed_sandbox_connection
+      :ok
+    end
+  end
+
+  defmodule SSHSandboxConnectionSuccess do
+    def session_channel(:sandbox, _, _, _), do: {:ok, 0}
+
+    def exec(_, _, _, _) do
+      send self(), :exec_sandbox_connection
+      {:ok, :result}
+    end
+  end
+
+  defmodule SSHSandboxConnectionError do
+    def session_channel(:sandbox, _, _, _), do: {:ok, 0}
+
+    def exec(_, _, _, _) do
+      send self(), :exec_sandbox_connection
+      :failure
+    end
+  end
+
+  setup_all do
+    {:ok, host: "foo.io"}
+  end
```

Comment:
i would expect this to rather be a module variable than a setup block (it doesn't really set things up.. it's more a constant you use in the tests)